### ISSUE TITLE
"Overlap error adjustment jobs failed, tried 2 times, giving up."

### ISSUE THREAD

**Issue by aseguret** [marbl/canu#1604]
Dear all,

Using the following command on Linux

canu -useGrid=false mhapMemory=60g mhapBlockSize=500 -p prefix -d directory genomeSize=315m corOutCoverage=200 correctedErrorRate=0.15 -pacbio-raw data.fastq

I receive the following output: 

```
-- BEGIN ASSEMBLY
--
-- No change in report.
-- No change in report.
--
-- Running jobs.  First attempt out of 2.
----------------------------------------
-- Starting 'oea' concurrent execution on Tue Jan 21 17:51:19 2020 with 2844.686 GB free disk space (1 processes; 64 concurrently)

    cd unitigging/3-overlapErrorAdjustment
    ./oea.sh 3 > ./oea.000003.out 2>&1

-- Finished on Tue Jan 21 17:51:30 2020 (11 seconds) with 2844.686 GB free disk space
----------------------------------------
--
-- Overlap error adjustment jobs failed, retry.
--   job 00003.oea FAILED.
--
-- No change in report.
--
-- Running jobs.  Second attempt out of 2.
----------------------------------------
-- Starting 'oea' concurrent execution on Tue Jan 21 17:51:30 2020 with 2844.686 GB free disk space (1 processes; 64 concurrently)

    cd unitigging/3-overlapErrorAdjustment
    ./oea.sh 3 > ./oea.000003.out 2>&1

-- Finished on Tue Jan 21 17:51:36 2020 (6 seconds) with 2844.686 GB free disk space
----------------------------------------
--
-- Overlap error adjustment jobs failed, tried 2 times, giving up.
--   job 00003.oea FAILED.
--

ABORT:
ABORT: Canu snapshot v2.0-development +249 changes (r9742 c64bf88508525cf4d74ab27092214bbea2b212b7)
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
```

When looking into this ./oea.000003.out file, the following error message can be found:

```
Running job 3 based on command line options.
Initializing.
Opening seqStore '../../tpila_canu.seqStore'.
sqStore_loadMetadata()-- Using 'corrected-trimmed' 0x10 reads.
Correcting reads 69532 to 105822.
Reading 66223212 corrections from './red.red'.
Correcting 333114292 bases with 56696448 indel adjustments.
--Allocate 317 + 432 + 1 MB for bases, adjusts and reads.
Corrected 299996208 bases with 203483 substitutions, 572562 deletions and 828748 insertions.
Loading overlaps.
Read_Olaps()--  Loading 8967189 overlaps from '../tpila_canu.ovlStore' for reads 69532 to 105822
--Allocate 273 MB for overlaps.
Read_Olaps()--  Loaded 8967189 overlaps -- 4513329 normal and 4453860 innie.
Sorting overlaps.

Failed with 'Segmentation fault'; backtrace (libbacktrace):
utility/system-stackTrace.C::89 in _Z17AS_UTL_catchCrashiP9siginfo_tPv()
(null)::0 in (null)()
```

It should be noted that this same version of CANU, using the same options, completed the assembly with another very similar genome recently on our system. I suppose it could be that this development version 2.0 is not stable, and that I should download the most recent updates, or the latest release (v1.9). I was wondering if you might have any other suggestions, as switching to another version would mean I need to rerun everything from the beginning (it ran for 1.5 months).

Many thanks for any help!

**Comment by skoren** [comment]
This is likely a bug that was fixed since your intermediate checkout, it's why we don't recommend running non-release versions for a production assembly. You could try to update to the latest in GitHub and see if the binary stores are still compatible with your assembly by re-starting. It's possible they are not as some of them have definitely changed since 1.9, in which case Canu will give a version error. If that doesn't work you'll have to install 1.9 and re-start the assembly step from scratch:
```
canu -assemble -useGrid=false -p prefix -d directory2 genomeSize=315m correctedErrorRate=0.15 -pacbio-corrected directory2/prefix.trimmedReads.fasta.gz
```

The reason the run took a while is likely a combination of running on a single node and setting the error rate to 15% along with 200x corrected reads. We don't normally run any pacbio datasets over 10% error and usually much lower. Is there a reason you set it so high here?

**Comment by aseguret** [comment]
Many thanks for your helpful reply. I appreciate that running non-release versions is not recommended, this was due to inattention on my part (until recently, I thought I had installed version 1.9).

I will try installing 1.9 and re-running the assembly without the correction and trimming, as you recommend. 

Last time, I encountered problems when trying to run it on the grid engine, but I will try this again to increase the number of nodes and speed up the process.

The reason I set the error rate so high was to smash haplotypes together (as suggested on [this page](<https://canu.readthedocs.io/en/latest/faq.html#what-parameters-can-i-tweak>), section What parameters can I tweak? -> For polyploid genomes -> Smash haplotypes together.
